### PR TITLE
Add string-concat-to-template codemod

### PR DESCRIPTION
**Pull request by dbrockman**
Transform string concatenations to template literals.

TODO:
- [x] Do not transform string concatenations if it only contains string literals.
- [ ] More test fixtures. Are there any edge cases?
- [ ] Support nested expressions like `'a' + (1 + 2) + 'b'`

### CODE REVIEW

File: transforms/string-concat-to-template.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  }
+  else {
+    inorderTreeTraversal(node.left, visit);
+    inorderTreeTraversal(node.right, visit);
+  }
+}
+
+
+function isConcatenation(node) {
+  return !!node && node.type === 'BinaryExpression' && node.operator === '+';
```
Comment (dbrockman) [inline transforms/string-concat-to-template.js]: - [ ] `isConcatenation` should check that at least one of `left` or `right` is a string literal
- [ ] Rename to `isStringConcatenation`

**Comment by dbrockman** [comment]
The expression `'a' + (1 + 2) + 'b'` produces the AST:

```json
{
    "type": "BinaryExpression",
    "operator": "+",
    "left": {
        "type": "BinaryExpression",
        "operator": "+",
        "left": {
            "type": "Literal",
            "value": "a",
            "raw": "'a'"
        },
        "right": {
            "type": "BinaryExpression",
            "operator": "+",
            "left": {
                "type": "Literal",
                "value": 1,
                "raw": "1"
            },
            "right": {
                "type": "Literal",
                "value": 2,
                "raw": "2"
            }
        }
    },
    "right": {
        "type": "Literal",
        "value": "b",
        "raw": "'b'"
    }
}
```

If a `BinaryExpression` is treated as any other expression if none of the sides are string literals then this should result in the template `a${1 + 2}b`
